x/emissions/keeper/msgserver/msg_server_stake.go

```diff
@@ -118,22 +110,17 @@ func (ms msgServer) ConfirmRemoveStake(ctx context.Context, msg *types.MsgConfir
 	if stakeRemoval.BlockRemovalStarted+delayWindow < currentBlock {
 		return nil, types.ErrConfirmRemoveStakeTooLate
 	}
-	// skip checking all the data is valid
-	// the data should be valid because it was checked when the stake removal was started
-	// send the money
-	for _, stakePlacement := range stakeRemoval.Placements {
-		// 5. check the module has enough funds to send back to the sender
-		// bank module does this for us in module SendCoins / subUnlockedCoins so we don't need to check
-		// 6. send the funds
-		amountInt := cosmosMath.NewIntFromBigInt(stakePlacement.Amount.BigInt())
-		coins := sdk.NewCoins(sdk.NewCoin(params.DefaultBondDenom, amountInt))
-		ms.k.SendCoinsFromModuleToAccount(ctx, types.AlloraStakingAccountName, senderAddr, coins)
+	// Check the module has enough funds to send back to the sender
+	// Bank module does this for us in module SendCoins / subUnlockedCoins so we don't need to check
+	// Send the funds
+	amountInt := cosmosMath.NewIntFromBigInt(stakeRemoval.Placement.Amount.BigInt())
+	coins := sdk.NewCoins(sdk.NewCoin(params.DefaultBondDenom, amountInt))
+	ms.k.SendCoinsFromModuleToAccount(ctx, types.AlloraStakingAccountName, sender, coins)
 
-		// 7. update the stake data structures
-		err = ms.k.RemoveStake(ctx, stakePlacement.TopicId, senderAddr, stakePlacement.Amount)
-		if err != nil {
-			return nil, err
-		}
+	// Update the stake data structures
+	err = ms.k.RemoveStake(ctx, stakeRemoval.Placement.TopicId, sender, stakeRemoval.Placement.Amount)
```

Comment:
Do we want to set `SetStakeRemoval` to empty after fully removing the stake?

Reply:
think you mean "set the map `stakeRemoval`/`delegateStakeRemoval` to empty after fully removing the stake", to which I'd say "yes"